extensions/amazon-aws/src/ecs.tsx

```diff
@@ -1,62 +1,35 @@
-import { Cluster, DescribeClustersCommand, ECSClient, ListClustersCommand } from "@aws-sdk/client-ecs";
-import { ActionPanel, List, Action, Icon } from "@raycast/api";
+import { Icon, List } from "@raycast/api";
 import { useCachedPromise } from "@raycast/utils";
-import AWSProfileDropdown, { AWS_URL_BASE } from "./aws-profile-dropdown";
+import { fetchClusters } from "./actions";
+import AWSProfileDropdown from "./components/searchbar/aws-profile-dropdown";
+import { getFilterPlaceholder } from "./util";
+import ECSCluster from "./components/ecs/ECSCluster";
+import { useState } from "react";
 
 export default function ECS() {
-  const { data: clusters, error, isLoading, revalidate } = useCachedPromise(fetchClusters);
+  const [isServicesLoading, setServicesLoading] = useState<boolean>(false);
```

Comment:
I don't think it's necessary to keep two loading states in this component. I would argue the `isLoading` from our `fetchClusters` is sufficient. The loading of each cluster should be handled by the cluster list item itself (maybe a loading indicator as the `icon`).

Furthermore, the `setServicesLoading` is passed inside a `.map` which means it will be called by each cluster and will lead to those clusters overwriting each other.

Reply:
I agree, good point, replaced each with loading icon if not complete as a replacement.